src/main/java/jenkins/plugins/slack/SlackNotifier.java

```diff
@@ -712,12 +815,18 @@ public FormValidation doTestConnection(@QueryParameter("baseUrl") final String b
                     targetUrl = this.baseUrl;
                 }
                 String targetDomain = Util.fixEmpty(teamDomain) != null ? teamDomain : this.teamDomain;
-                boolean targetBotUser = botUser || this.botUser;
-                String targetTokenCredentialId = Util.fixEmpty(tokenCredentialId) != null ? tokenCredentialId :
```

Comment:
removing these two lines break relying on global config when testing in a freestyle job

Reply:
You're right, the test needs to be consistent so the behavior is the same between the test and how it actually works.